### PR TITLE
Latest Embeddings => Spaces/Embeddings branch

### PR DESCRIPTION
**Pull request by ritch**

### CODE REVIEW

File: app/packages/core/src/components/SamplesContainer.tsx [hunk 1]
```diff
@@ -151,10 +151,7 @@ function SamplesContainer() {
   return (
     <Container>
       {showSidebar && <Sidebar render={renderGridEntry} modal={false} />}
-      <SpacesRoot
-        id={`primary-${datasetName}`}
-        defaultState={defaultSpacesState}
-      />
+      <SpacesRoot id="primary" defaultState={defaultSpacesState} />
```
Comment (imanjra) [inline app/packages/core/src/components/SamplesContainer.tsx:154]: Is this change intentional?

File: app/packages/core/src/components/ViewBar/ViewStage/ViewStageParameter.tsx [hunk 2]
```diff
@@ -215,6 +215,7 @@ const ObjectEditor = ({
   const attach = () => {
     request && window.cancelAnimationFrame(request);
     request = window.requestAnimationFrame(() => {
+      if (!containerRef.current) return;
```
Comment (imanjra) [inline app/packages/core/src/components/ViewBar/ViewStage/ViewStageParameter.tsx:218]: Nice!

File: app/packages/embeddings/src/Embeddings.tsx [hunk 3]
```diff
@@ -1,31 +1,49 @@
-import { useEffect, useState, useRef, useLayoutEffect } from "react";
-import { useRecoilState, useRecoilValue, useRecoilCallback } from "recoil";
+import { useEffect, useState, useRef, useLayoutEffect, Fragment } from "react";
+import {
+  useRecoilState,
+  useRecoilValue,
+  useRecoilCallback,
+  atom,
```
Comment (imanjra) [inline app/packages/embeddings/src/Embeddings.tsx:6]: Is this used?

File: app/packages/embeddings/src/Embeddings.tsx [hunk 4]
```diff
@@ -1,31 +1,49 @@
...
+} from "recoil";
 import * as fos from "@fiftyone/state";
-import { getFetchFunction } from "@fiftyone/utilities";
+import { getFetchFunction, useExternalLink } from "@fiftyone/utilities";
 import Plot from "react-plotly.js";
-import { Button, Loading, Selector } from "@fiftyone/components";
+import { Loading, Selector, useTheme, Link } from "@fiftyone/components";
 import styled from "styled-components";
-import { State } from "@fiftyone/state";
+import { useToPatches } from "@fiftyone/state";
```
Comment (imanjra) [inline app/packages/embeddings/src/Embeddings.tsx:13]: Is this used?

File: app/packages/embeddings/src/Embeddings.tsx [hunk 5]
```diff
@@ -1,31 +1,49 @@
...
+import { usePanelState, usePanelStatePartial } from "@fiftyone/spaces";
```
Comment (imanjra) [inline app/packages/embeddings/src/Embeddings.tsx:14]: Is `usePanelState` used?

File: app/packages/embeddings/src/Embeddings.tsx [hunk 6]
```diff
@@ -1,31 +1,49 @@
-import { useEffect, useState, useRef, useLayoutEffect } from "react";
-import { useRecoilState, useRecoilValue, useRecoilCallback } from "recoil";
+import { useEffect, useState, useRef, useLayoutEffect, Fragment } from "react";
```
Comment (imanjra) [inline app/packages/embeddings/src/Embeddings.tsx:1]: Looks like `useLayoutEffect` is no longer used

File: app/packages/embeddings/src/Embeddings.tsx [hunk 7]
```diff
@@ -37,114 +55,311 @@ function useBrainResultsSelector() {
...
-  width: 10rem;
   display: flex;
   gap: 1rem;
+  position: absolute;
+  top: 1rem;
+  display: flex;
+  column-gap: 1rem;
+  z-index: 999;
+  padding: 0 1rem;
+  // width: 100%;
```
Comment (imanjra) [inline app/packages/embeddings/src/Embeddings.tsx:163]: Can these be removed?

File: app/packages/embeddings/src/Embeddings.tsx [hunk 8]
```diff
@@ -174,105 +389,171 @@ function findIndexByKeyValue(array, key, value) {
...
       //   .filter((d) => d !== null);
-      const selectedpoints = plotSelection.length
-        ? plotSelection.map((id) => findIndexByKeyValue(trace, "id", id))
+      const selectedpoints = plotSelection?.length
+        ? plotSelection
+            .map((id) => getPointIndex(trace, id))
+            .filter((p) => p !== null)
         : null;
 
+      // const color = Color.fromCSSRGBValues(r, g, b)
```
Comment (imanjra) [inline app/packages/embeddings/src/Embeddings.tsx:418]: Can comment be removed?

File: app/packages/core/src/components/SamplesContainer.tsx (hunk 1 above)
Comment (ritch) [reply app/packages/core/src/components/SamplesContainer.tsx:154]: No - reverting...
